pkg/api/customization/cluster/formatter.go

```diff
@@ -60,8 +68,8 @@ func (f *Formatter) Formatter(request *types.APIContext, resource *types.RawReso
 				resource.Schema) == nil
 		}
 		canUpdateCluster := canUpdateClusterFn(request, resource.ID)
-		logrus.Debugf("user: %v, canUpdateCluster: %v", request.Request.Header.Get("Impersonate-User"), canUpdateCluster)
-		if canUpdateCluster {
+		logrus.Debugf("isActiveCluster: %v isWindowsCluster: %v user: %v, canUpdateCluster: %v", isActiveCluster, isWindowsCluster, request.Request.Header.Get("Impersonate-User"), canUpdateCluster)
```

Comment:
canUpdateClusterFn is an extra call, we should skip it after checking isActive  and isWindows

Reply:
We need to check RBAC before doing anything else. Otherwise we would be leaking information.

Reply:
This is just formatter, so I don't think we'd be leaking information here by checking RBAC only if cluster is active and not windows

Reply:
oops, you are right. I was looking at actions_cis.go file in Goland, will change.